subsys/bluetooth/controller/ll_sw/ull_llcp_phy.c

```diff
@@ -904,38 +940,53 @@ bool llcp_lp_pu_awaiting_instant(struct proc_ctx *ctx)
 /*
  * LLCP Remote Procedure PHY Update FSM
  */
-static void rp_pu_tx(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t opcode)
+static void rp_pu_tx(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
 {
 	struct node_tx *tx;
 	struct pdu_data *pdu;
 
-	/* Allocate tx node */
-	tx = llcp_tx_alloc(conn, ctx);
-	LL_ASSERT(tx);
+	/* (pre)allocate tx node, but only do it if not already done */
+	if (ctx->node_ref.tx_ack == NULL) {
+		ctx->node_ref.tx_ack = llcp_tx_alloc(conn, ctx);
+		LL_ASSERT(ctx->node_ref.tx_ack);
+	}
```

Comment:
For the case of Central Role, doesn't the tx node allocate for the LL_PHY_REQ get reused for the LL_PHY_UPDATE_IND as tx_ack is never nulled, or i'm missing something...

Reply:
the reference in tx_ack is null'ed on TX ack handling

Reply:
Good then I was missing something